observationtree/src/main/scala/explore/observationtree/TargetObsList.scala

```diff
@@ -176,24 +177,26 @@ object TargetObsList {
         targetMod(setter, props.targetsWithObs, props.focused, newTarget.id)(upsert).runInCB
       }
 
-    protected def deleteTargetEnabled(
-      focused:        Option[Focused],
-      targetsWithObs: TargetsWithObs
-    ): Option[Target.Id] =
-      focused.collect {
-        case FocusedTarget(targetId) if !targetsWithObs.obs.exists(_.target.id === targetId) =>
-          targetId
-      }
-
-    protected def deleteTarget(setter: Undoer.Setter[IO, TargetsWithObs]): Callback =
-      $.props >>= { props =>
-        deleteTargetEnabled(props.focused.get, props.targetsWithObs.get)
-          .map(targetId =>
-            targetMod(setter, props.targetsWithObs, props.focused, targetId)(
-              targetListMod.delete
-            ).runInCB
-          )
-          .getOrEmpty
+    protected def deleteTarget(
```

Comment:
@rpiaggio I struggled a bit with this part, it maybe worth a review

Reply:
I admit all this original code lacks a lot of comments.